### DISCUSSION TITLE
Mack random normal distribution for each origin period

### DISCUSSION THREAD

**Discussion by rbreen2** [casact/chainladder-python#212] [Q&A]
In the tutorial we see how we can use the ibnr sum as the mean and the total mack std error as the sd to generate a normal distribution. If we wanted to get a normal dist of the Ultimate for each origin period using the data from the mack.summary_ would we use the ultimate and its corresponding mack std error for a specific origin period as the parameters for np.random.normal?

**Comment by kennethshsu** [comment]
Yes, one easy way is to use the `.summary_` call on the Mack estimator.

```python
import pandas as pd
import numpy as np
import chainladder as cl

clrd = (
    cl.load_sample("clrd")
    .groupby("LOB")
    .sum()
    .loc["wkcomp", ["CumPaidLoss", "EarnedPremNet"]]
)
mack = cl.MackChainladder().fit(clrd["CumPaidLoss"])
mack.summary_
```

`summary_` will print the `Latest`, `IBNR`, `Ultimate`, and `Mack Std Err` for each origin. 

If we want to reference only a specific origin, for example 1990, we can do this:
```python
print(
    "Mack's mean ultimate for origin 1990:",
    mack.summary_.to_frame().loc["1990"]["Ultimate"],
)
print(
    "Mack's standard error for origin 1990:",
    mack.summary_.to_frame().loc["1990"]["Mack Std Err"],
)
```

**Reply by rbreen2** [reply]
Yes, so would we be allowed to use those parameters and do something similar in the tutorial to get a normal distribution of the ultimate for that year? 

`ult_mean = mack.summary_.to_frame().loc["1990"]["Ultimate"]
ult_sd = mack.summary_.to_frame().loc["1990"]["Mack Std Err"]
n_trials = 10000

np.random.seed(2021)
dist = np.random.normal(ult_mean, ult_sd, size=n_trials)

plt.hist(dist, bins=50)`

Or is the mean suppose to be the IBNR with Mack Std Err as the standard deviation? Which would then give us the IBNR normal distribution for the year.

**Reply by kennethshsu** [reply]
The standard error is for the IBNR. But since there's no variance in `Latest` (everything is known to date), so adding a constant to the simulated distribution is mathematically sound. We are basically shifting our distribution up by a constant, `Latest`.

**Reply by rbreen2** [reply]
Thank you, that's what I was thinking but got confused when in the first origin period where the IBNR is 0, there was a Mack Std Err. So then using the Ultimate which really is just the 'latest' would produce some normal distribution.

**Reply by kennethshsu** [reply]
Right, I think a big assumption here is that the IBNR estimates are normally distributed. Depends on the context, it may or may not be violated. One should always be cautious of making such assumption. For this specific scenario, the first origin's IBNR is estimated with the mean of 0 and some Std Error, this will imply possible downward development, which might not make sense at all.
